src/Content/Text/BBCode.php

```diff
@@ -636,4 +646,1341 @@ public static function limitBodySize($body)
+			}
+
+			$orig_body = substr($orig_body, $img_end + strlen('[/img]'));
+
+			if ($orig_body === false) {
+				// in case the body ends on a closing image tag
+				$orig_body = '';
+			}
+
+			$img_start = strpos($orig_body, '[img');
+			$img_st_close = ($img_start !== false ? strpos(substr($orig_body, $img_start), ']') : false);
+			$img_end = ($img_start !== false ? strpos(substr($orig_body, $img_start), '[/img]') : false);
+		}
+
+		$new_body = $new_body . $orig_body;
+
+		return ['body' => $new_body, 'images' => $saved_image];
+	}
+
+	private static function interpolateSavedImagesIntoItemBody($body, array $images)
+	{
+		$newbody = $body;
+
+		$cnt = 0;
+		foreach ($images as $image) {
+			// We're depending on the property of 'foreach' (specified on the PHP website) that
+			// it loops over the array starting from the first element and going sequentially
+			// to the last element
+			$newbody = str_replace('[$#saved_image' . $cnt . '#$]',
+				'<img src="' . proxy_url($image) . '" alt="' . L10n::t('Image/photo') . '" />', $newbody);
+			$cnt++;
+		}
+
+		return $newbody;
+	}
+
+	/**
+	 * Processes [share] tags
+	 *
+	 * Note: Can produce a [bookmark] tag in the output
+	 *
+	 * @brief Processes [share] tags
+	 * @param array    $share      preg_match_callback result array
+	 * @param bool|int $simplehtml
+	 * @return string
+	 */
+	private static function convertShare($share, $simplehtml)
+	{
+		$attributes = $share[2];
+
+		$author = "";
+		preg_match("/author='(.*?)'/ism", $attributes, $matches);
+		if (x($matches, 1)) {
+			$author = html_entity_decode($matches[1], ENT_QUOTES, 'UTF-8');
+		}
+
+		preg_match('/author="(.*?)"/ism', $attributes, $matches);
+		if (x($matches, 1)) {
+			$author = $matches[1];
+		}
+
+		$profile = "";
+		preg_match("/profile='(.*?)'/ism", $attributes, $matches);
+		if (x($matches, 1)) {
+			$profile = $matches[1];
+		}
+
+		preg_match('/profile="(.*?)"/ism', $attributes, $matches);
+		if (x($matches, 1)) {
+			$profile = $matches[1];
+		}
+
+		$avatar = "";
+		preg_match("/avatar='(.*?)'/ism", $attributes, $matches);
+		if (x($matches, 1)) {
+			$avatar = $matches[1];
+		}
+
+		preg_match('/avatar="(.*?)"/ism', $attributes, $matches);
+		if (x($matches, 1)) {
+			$avatar = $matches[1];
+		}
+
+		$link = "";
+		preg_match("/link='(.*?)'/ism", $attributes, $matches);
+		if (x($matches, 1)) {
+			$link = $matches[1];
+		}
+
+		preg_match('/link="(.*?)"/ism', $attributes, $matches);
+		if (x($matches, 1)) {
+			$link = $matches[1];
+		}
+
+		$posted = "";
+
+		preg_match("/posted='(.*?)'/ism", $attributes, $matches);
+		if (x($matches, 1)) {
+			$posted = $matches[1];
+		}
+
+		preg_match('/posted="(.*?)"/ism', $attributes, $matches);
+		if (x($matches, 1)) {
+			$posted = $matches[1];
+		}
+
+		// We only call this so that a previously unknown contact can be added.
+		// This is important for the function "get_contact_details_by_url".
```

Comment:
The `get_contact...` function now has a different name, shoud be updated in the comment as well.

Reply:
Done.